### DISCUSSION TITLE
能否在低代码Pipeline中调用一个由Python实现的CustomAction？

### DISCUSSION THREAD

**Discussion by lumina37** [MaaXYZ/MaaFramework#436] [Q&A] [answered]
最近我们想用Py实现一个比较复杂的Action，但还想沿用之前的MFAWPF作为GUI。我粗略阅读了Py相关的文档，发现demo的实现方式是从Py启动并调用Py中自定义的Action。因此想请教一下能否在低代码Pipeline中调用一个由Python实现的CustomAction？

**Comment by dongwlin** [comment]
pipeline可以调用python实现的custom action。但是mfawpf是用c#实现的，你不能为它提供python实现的custom action。

**Reply by lumina37** [reply]
好的感谢回复😘

**Comment by MistEO** [comment]
等 pi server 做出来可能就有了（画大饼

https://github.com/MaaXYZ/MaaFramework/issues/333

**Comment by Windsland52** [comment]
lol

**Comment by weinibuliu** [comment, marked as answer]
4.0 版本已实现通过 Agent 注册与调用 Custom
